src/longrange_force_types/force_ewald_cabana_neigh_impl.h

```diff
@@ -0,0 +1,597 @@
+                x( idx, dim ) -= domain_size( dim );
+            }
+        }
+
+        offset += halos.at( 2 * dim + 1 )->numGhost();
+    }
+
+    // create VerletList to iterate over
+
+    double min_coords[3];
+    double max_coords[3];
+
+    for ( int dim = 0; dim < 3; ++dim )
+    {
+        max_coords[dim] = -2.0 * r_max;
+        min_coords[dim] = domain_size( dim ) + 2.0 * r_max;
+    }
+
+    for ( int idx = 0; idx < N_max; ++idx )
+    {
+        for ( int dim = 0; dim < 3; ++dim )
+        {
+            if ( x( idx, dim ) < min_coords[dim] )
+                min_coords[dim] = x( idx, dim );
+            if ( x( idx, dim ) > max_coords[dim] )
+                max_coords[dim] = x( idx, dim );
+        }
+    }
+
+    // compute cell size in a way that not too many cells are used
+    double cell_size;
+    cell_size =
+        std::max( std::min( ( grid_max[0] - grid_min[0] ) / 20.0,
+                            std::min( ( grid_max[1] - grid_min[1] ) / 20.0,
+                                      ( grid_max[2] - grid_min[2] ) / 20.0 ) ),
+                  1.0 );
+
+    ListType verlet_list( x, 0, N_local, r_max, cell_size, grid_min, grid_max );
+
+    Kokkos::parallel_for( N_local, KOKKOS_LAMBDA( const int idx ) {
+        int num_n =
+            Cabana::NeighborList<ListType>::numNeighbor( verlet_list, idx );
+
+        double rx = x( idx, 0 );
+        double ry = x( idx, 1 );
+        double rz = x( idx, 2 );
+
+        for ( int ij = 0; ij < num_n; ++ij )
+        {
+            int j = Cabana::NeighborList<ListType>::getNeighbor( verlet_list,
+                                                                 idx, ij );
+            double dx = x( j, 0 ) - rx;
+            double dy = x( j, 1 ) - ry;
+            double dz = x( j, 2 ) - rz;
+            double d = sqrt( dx * dx + dy * dy + dz * dz );
+
+            // potential computation
+            double contrib = 0.5 * q( idx ) * q( j ) * erfc( alpha * d ) / d;
+            Kokkos::atomic_add( &p( idx ), contrib );
+            Kokkos::atomic_add( &p( j ), contrib );
+
+            // force computation
+           double f_fact = q( idx ) * q( j ) *
+                            ( 2.0 * sqrt( alpha / PI ) * exp( -alpha * d * d ) +
+                              erfc( sqrt( alpha ) * d ) ) /
+                            ( d * d );
+            Kokkos::atomic_add( &f( idx, 0 ), f_fact * dx );
+            Kokkos::atomic_add( &f( idx, 1 ), f_fact * dy );
+            Kokkos::atomic_add( &f( idx, 2 ), f_fact * dz );
+            Kokkos::atomic_add( &f( j, 0 ), -f_fact * dx );
+            Kokkos::atomic_add( &f( j, 1 ), -f_fact * dy );
+            Kokkos::atomic_add( &f( j, 2 ), -f_fact * dz );
+        }
+    } );
+    Kokkos::fence();
+
+
+    // send the force and potential contributions of the
+    // ghost particles back to the origin processes
+    for ( int n_halo = 5; n_halo >= 0; --n_halo )
+    {
+        Cabana::scatter( *( halos.at( n_halo ) ), p );
```

Comment:
Same as above about keeping comm out of this force, except that it may be reasonable to use a full list for short range (where you don't need to scatter) and half list for long range